client/src/parsec/classes.ts

```diff
@@ -0,0 +1,469 @@
+  }
+
+  _assertState(nullCanceller: boolean, nullHandle: boolean): void {
+    if (nullCanceller && this.canceller !== null) {
+      throw Error('Canceller should be null');
+    } else if (!nullCanceller && this.canceller === null) {
+      throw Error('Canceller should not be null');
+    }
+    if (nullHandle && this.handle !== null) {
+      throw Error('Handle should be null');
+    } else if (!nullHandle && this.handle === null) {
+      throw Error('Handle should not be null');
+    }
+  }
+
+  async retrieveInfo(invitationLink: string):
+  Promise<Result<UserOrDeviceClaimInitialInfoUser, ClaimerRetrieveInfoError>> {
+    function eventCallback(event: ClientEvent): void {
+      console.log('On event', event);
+    }
+
+    this._assertState(true, true);
+
+    if (window.isDesktop()) {
+      const clientConfig = getClientConfig();
+      const result = await libparsec.claimerRetrieveInfo(clientConfig, eventCallback, invitationLink);
+      if (result.ok) {
+        this.handle = result.value.handle;
+        this.greeter = result.value.greeterHumanHandle;
+      }
+      return result as Result<UserOrDeviceClaimInitialInfoUser, ClaimerRetrieveInfoError>;
+    } else {
+      await _wait(this.MOCK_HOST_WAITING_TIME);
+      this.handle = DEFAULT_HANDLE;
+      this.greeter = {
+        email: 'gale@waterdeep.faerun',
+        // cspell:disable-next-line
+        label: 'Gale Dekarios',
+      };
+      return new Promise<Result<UserOrDeviceClaimInitialInfoUser, ClaimerRetrieveInfoError>>((resolve, _reject) => {
+        resolve({ok: true, value: {
+          tag: 'User',
+          handle: DEFAULT_HANDLE,
+          claimerEmail: 'shadowheart@swordcoast.faerun',
+          greeterUserId: '1234',
+          greeterHumanHandle: {
+            email: 'gale@waterdeep.faerun',
+            // cspell:disable-next-line
+            label: 'Gale Dekarios',
+          },
+        }});
+      });
+    }
+  }
+
+  async initialWaitHost():
+  Promise<Result<DeviceClaimInProgress1Info, ClaimInProgressError>> {
+    this._assertState(true, false);
+    if (window.isDesktop()) {
+      this.canceller = await libparsec.newCanceller();
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const result = await libparsec.claimerDeviceInitialDoWaitPeer(this.canceller, this.handle!);
+      if (result.ok) {
+        this.SASCodeChoices = result.value.greeterSasChoices;
+        this.correctSASCode = result.value.greeterSas;
+        this.handle = result.value.handle;
+      }
+      this.canceller = null;
+      return result;
+    } else {
+      this.SASCodeChoices = ['5MNO', '6PQR', '7STU', '8VWX'];
+      this.correctSASCode = '7STU';
+      return new Promise<Result<DeviceClaimInProgress1Info, ClaimInProgressError>>((resolve, _reject) => {
+        resolve({ok: true, value: {
+          handle: DEFAULT_HANDLE,
+          greeterSas: '7STU',
+          greeterSasChoices: ['5MNO', '6PQR', '7STU', '8VWX'],
```

Comment:
Might as well use the variables, this avoid declaring the values twice (and avoid potential bugs if it's changed later).